Jonathan's SPARQL problems

jonathanvajda opened the pull request:
See reviews from Giacomo and Jaron

jonathanvajda committed:
Typo fix

The project was copy-and-pasted, but not all deets were updated to reflect the project is the 2nd one

jonathanvajda committed:
Merge pull request #1 from jonathanvajda/jonathanvajda-patch-1

Typo fix
Not best practice to accept one's own pull request

jonathanvajda committed:
Merge remote-tracking branch 'upstream/main'

jonathanvajda committed:
Merge branch 'johnbeve:main' into main

jonathanvajda committed:
Create new file

This is my attempt. I have only a handful that need their SPARQL queries fixed (they are obviously broken). Those are marked with "borked" in the body.

I will adjust those in next 24 hrs. Any PRs welcome.

jonathanvajda committed:
Revert "Create new file"

This reverts commit 0a0b2e621ed72ebdef9c40cf3a478463af7bb41e.

jonathanvajda committed:
Create SPARQL-JV.md

This is my attempt. I have only a handful that need their SPARQL queries fixed (they are obviously broken). Those are marked with "borked" in the body.

I will adjust those in next 24 hrs. Any PRs welcome.

jonathanvajda committed:
updated sparql queries

Tweaked 7-1, 7-3
Fixed 7-8
Realized 7-9 is actually more difficult, and a Kata-6 (now Problem 6-7)

Still have to fix:
6-3, 6-6
5-3, 5-4, 5-5
4-2, 4-4

Problem 4-2 should probably be harder. I have a plan.

jonathanvajda committed:
Update SPARQL-JV.md

Tweaked 7-7
Fixed 6-3
Fixed 6-6

Remaining:
5-3, 5-4, 5-5
4-2, 4-4

Problem 4-2 should probably be harder. I have a plan.

jonathanvajda committed:
Update SPARQL-JV.md

fixed code format

jonathanvajda committed:
Update SPARQL-JV.md

Fixed 5-3
Swapped 4-2 (became 5-4) and 5-4 (became 4-2)
Fixed 5-4
Fixed 4-2

Still need to fix 5-5 and 4-4

jonathanvajda committed:
updates to a pile of queries

need to fix 5-5 and 4-4

jonathanvajda committed:
updated 5-5

I am not confident that the syntax is right for the negation of the regex. I am going for a negation of a disjunction

jonathanvajda committed:
Update SPARQL-JV.md

jonathanvajda committed:
Update SPARQL-JV.md

jonathanvajda committed:
Update SPARQL-JV.md

jonathanvajda committed:
Update SPARQL-JV.md

jonathanvajda committed:
4-1 fixed

It was so janky and the difficulty wasn't up to Kata-4, imo. I think it is sufficiently difficult now.

jonathanvajda committed:
Penultimate version I believe

Finally fixed all of them to the best of my ability, sans others' feedback.

jonathanvajda committed:
addressing review objections

Giacomo gave feedback that this would be helpful. I think it would be helpful for all of the prompts, but I will probably not be able to furnish them all in time.

jonathanvajda committed:
fixed age query

I was making it too difficult to filter by age. Fixed!

jonathanvajda committed:
restated ?device with semicolon

Giacomo's comment was right. Fixed!

jonathanvajda committed:
6-1 too easy?

Giacomo said it seemed too easy. I agree. I think it is better now.

jonathanvajda committed:
typo on primarily_composed_of

fixed primarily composed of

jonathanvajda committed:
5-2 was too easy Giacomo says...

I don't want it to be a kata 6, so I have ramped up the expectations by requiring a CONSTRUCT

jonathanvajda committed:
Update SPARQL-JV.md

raising the number of operators or functions for my Kata-7's, per Jaron's recommendation

jonathanvajda committed:
Update SPARQL-JV.md

Hopefully I have raised the expectations

jonathanvajda committed:
Merge pull request #2 from jonathanvajda/jonathanvajda-sparql-problems

Create SPARQL-JV.md

giacomodecolle commented on Project 4/SPARQL-JV.md:
````diff
@@ -0,0 +1,526 @@
+- The FBI wants to put several people on a watchlist if they have performed some suspicious activity. Suppose that they wanted to put anyone on the watchlist who fit the following conditions: the person owns a phone that was present at the US Capital Building on January 6, 2022. There are some ways to connect the device to the building by finding such cases where the mobile device has certain GPS ...
+- Assert with CONSTRUCT that such persons bear a FBIWatchlistMemberRole
+- NOTE: Feel free to tell me that this is a Kata-3?
+```
````
That's good, unfortunately I am not sure whether it can be a level 3 or not. The way in which the query is set up is for sure complex, and you do use multiple "tricks" (filters, etc.) but there is still no consensus on what distinguishes a level 4 from a level 3.

giacomodecolle commented on Project 4/SPARQL-JV.md:
````diff
@@ -0,0 +1,526 @@
+- Find all specimens that are instances of blood or bone marrow, derived from a woman younger than 25 years old, who has not tested positive for cancer (and if there has been a negative test result for cancer, include that information), and the specimen is located in the state of Illinois, Indiana, Michigan, or Ohio.
+- NOTE 1: This is largely from ChatGPT, generated 2023-03-25
+- NOTE 2: I am kinda hoping you rate this as a Kata-3. Nudge.
+```
````
I am still not sure about the difference between level 3 and level 4. But this could be a level 3, more than the exercise below. I'm not sure how to express this, but I have the feeling that this kata requires a bit more lateral thinking from the user.

giacomodecolle commented on Project 4/SPARQL-JV.md:
````diff
@@ -0,0 +1,526 @@
+
+**Problem 4-2.**
+- An E. coli breakout has occurred recently and the culprit is contaminated iceberg lettuce pallets in the distribution for fastfood restaurants in the midwestern states. All pallets of lettuce that passed through distribution chains in Arkansas and Nebraska are at risk. Declare all such lettuce to bear an "E. coli risk" role and "recall" quality status, if they were present on shipping routes th ...
+```
````
Cool use of DELETE, you might be the first person I have seen using this. Would you consider saying more explicitly in the prompt that teh user needs to use DELETE?

giacomodecolle commented on Project 4/SPARQL-JV.md:
````diff
@@ -0,0 +1,526 @@
+```
+
+**Problem 5-5.**
+- Suppose that new research shows that males are at heightened risk for testicular cancer if they have the conjunction of two genes, erbB-2 (Receptor tyrosine-protein kinase erbB-2) and PMS2 (Mismatch repair endonuclease PMS2). However, those whose mitochondrial haplogroup L0, L1, L2, L4, L5, L6 (i.e., L1-L6, except L3) are not yet known to be significantly affected. Notifications must be sent ou ...
````
If you are interested into ontological representations of the receptor tyrosine kinase family, check out the PRC (protein conformation ontology), on which I am currently working on!

giacomodecolle commented on Project 4/SPARQL-JV.md:
````diff
@@ -0,0 +1,526 @@
+
+**Problem 5-5.**
+- Suppose that new research shows that males are at heightened risk for testicular cancer if they have the conjunction of two genes, erbB-2 (Receptor tyrosine-protein kinase erbB-2) and PMS2 (Mismatch repair endonuclease PMS2). However, those whose mitochondrial haplogroup L0, L1, L2, L4, L5, L6 (i.e., L1-L6, except L3) are not yet known to be significantly affected. Notifications must be sent ou ...
+```
````
Besides my self-adverstising, the query is good for a five

giacomodecolle commented on Project 4/SPARQL-JV.md:
````diff
@@ -0,0 +1,526 @@
+}
+```
+
+**Problem 5-3.**
````
with concat and the other functions used it is a solid level 5. I am myself ignorant about the syntax of CONSTRUCT REDUCED

giacomodecolle commented on Project 4/SPARQL-JV.md:
````diff
@@ -0,0 +1,526 @@
+
+**Problem 6-7.**
+- Find all musicians born in 1960s, who were members of some grunge band that was based in Seattle.
+```
````
6 is good

giacomodecolle commented on Project 4/SPARQL-JV.md:
```diff
@@ -0,0 +1,526 @@
+**The SPARQL Library of Buffalo**
+
```
I think I have reviewed all the new katas you added and confirmed the levels for most of them. Please let me know if I missed something - my memory of what I reviewed a couple of days ago is quite bad.

giacomodecolle reviewed (approved):
All good for me, see

JaronJCheung commented on Project 4/SPARQL-JV.md:
```diff
@@ -0,0 +1,526 @@
+**The SPARQL Library of Buffalo**
+
+[Codewars](https://www.codewars.com/dashboard) is a website designed to facilitate algorithmic training for various programming languages. Users supply problem statements and others provide coding solutions to those problems. For example, you might find a problem for Python such as: 
```
Dear @jonathanvajda,

The new changes look good! The kata 7s look on par and the rest of the changes that were suggested are a solid improvement from the last iteration. 

Well done! 

Thanks for inviting me to review your project!

Best, 
Jaron

jonathanvajda committed:
Noticed a design pattern mistake

I was missing type assertions for many variables. This is a problem that recurs throughout. Oops? More to correct, btw
